src/app/demo/demo.component.ts

```diff
@@ -27,7 +28,71 @@ export class DemoComponent implements OnInit {
 
   ngOnInit(): void {}
 
+  // click botton btn just once
+  clicked = false;
+
+  //addTranslationLine is called when the user clicks the "Add Translation" button
+  addTranslationLine(): void {
+    // @ts-ignore
+    const readalongRoot: any = document.querySelector("read-along").shadowRoot;
+    const sentences = readalongRoot.querySelectorAll(".sentence");
+    sentences.forEach((sentence: any) => {
+      const innerbutton = document.createElement("button");
+      innerbutton.innerHTML = "Button";
+      // innerbutton.setAttribute("click", "clicked = true");
+      innerbutton.addEventListener("click", () => {
+        sentence.insertAdjacentHTML(
+          "beforeend",
+          '<br><span class = "translation" contenteditable = True>Translation</span>'
+        );
+        innerbutton.disabled = true;
+      });
+      sentence.insertAdjacentElement("afterend", innerbutton);
+    });
+  }
```

Comment:
This function works well for the add translation line. It is good.

Reply:
Thanks for review.